runtime/interpreter/value_range.go

```diff
@@ -0,0 +1,208 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package interpreter
+
+import (
+	"fmt"
+
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/errors"
+	"github.com/onflow/cadence/runtime/sema"
+)
+
+// NewInclusiveRangeValue constructs an InclusiveRange value with the provided start, end with default value of step.
+func NewInclusiveRangeValue(
+	interpreter *Interpreter,
+	locationRange LocationRange,
+	start IntegerValue,
+	end IntegerValue,
+	rangeType InclusiveRangeStaticType,
+) *CompositeValue {
+	startComparable, startOk := start.(ComparableValue)
+	endInclusiveComparable, endInclusiveOk := end.(ComparableValue)
+	if !startOk || !endInclusiveOk {
+		panic(errors.NewUnreachableError())
+	}
+
+	step := interpreter.GetValueForIntegerType(1, rangeType.ElementType)
+	if startComparable.Greater(interpreter, endInclusiveComparable, locationRange) {
+		elemSemaTy := interpreter.MustConvertStaticToSemaType(rangeType.ElementType)
+		if _, ok := sema.AllUnsignedIntegerTypesSet[elemSemaTy]; ok {
```

Comment:
instead of the map lookup, perhaps use the `TypeTag`.
e.g: `if elemSemaTy.Tag().BelongsTo(UnsignedIntegerTypeTag) {} `

That way, we also don't need to introduce a `AllUnsignedIntegerTypesSet`

Reply:
Done, using tags now and removed the set.